homework02/allan/string.h

```diff
@@ -0,0 +1,41 @@
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+
+class String
+{
+public:
+    String();
+    String(const char *);
+    String(const String& other);
+                String(const char *, int capactiy);
+    String& operator=(String other);
+    ~String();
+
+    bool empty() const;
+    const char* c_str() const ;
+
+    operator std::string() const;
+
+    char& operator[](int pos);
+    const char& operator[](int pos) const;
```

Comment:
Good, although for types such as `char` it is better / simpler to return it by value rather than const ref